### DISCUSSION TITLE
Wildcard matching with "todo.sh list"

### DISCUSSION THREAD

**Discussion by richmintz1** [todotxt/todo.txt-cli#334] [Q&A] [answered]
Wildcard matching of the form ```todo.sh list abc*``` works, almost always exactly as I would expect, but I'm discovering a few quirks. Here's one: why are the first two commands matching item 156? Am I failing to see something obvious? [They both also, I think as intended, match on things like ```+1-NWX``` -- I tested.]

![image](https://user-images.githubusercontent.com/50835754/109075038-962bff00-76c6-11eb-9120-46625578e2d8.png)

**Comment by inkarkat** [comment, marked as answer]
This is because the _wildcard matching_... is in fact _regular expression_ matching, so `+1-NW*` is (as a basic regexp for `grep`) matching the literal `+1-N` followed by 0 or more `W` characters (which `+1-NRR` fulfills).

Matching is done case-insensitive with `grep`; the only specialty added is that a leading dash inverts the matching (i.e. `grep -v`); here's the [line in the code](https://github.com/todotxt/todo.txt-cli/blob/bf1bf1ae8423fffa395d70e79ba0385cdc4e1faf/todo.sh#L862).

Unfortunately, the `todo.sh help list` is very vague in that regard; it just speaks of _term(s)_; the only hint that basic regular expressions are used is in the fact that logical or is achieved via `\|`. I've just submitted https://github.com/todotxt/todo.txt-cli/pull/335 to improve that.
